sycl/test/on-device/online_compiler/online_compiler_common.hpp

```diff
@@ -0,0 +1,171 @@
+      IL = Compiler.compile(
+          std::string(CLSource),
+          // Intentionally use one option twice.
+          std::vector<std::string>{std::string("-cl-fast-relaxed-math"),
+                                   std::string("-cl-fast-relaxed-math")});
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+    testSyclKernel(Q, getSYCLKernelWithIL(Context, IL));
+  }
+
+  { // Compile and run a trivial OpenCL kernel using online_compiler()
+    // constructor accepting SYCL device.
+    std::cout << "Test case2\n";
+    online_compiler<source_language::opencl_c> Compiler(Device);
+    std::vector<byte> IL;
+    try {
+      IL = Compiler.compile(std::string(CLSource));
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+    testSyclKernel(Q, getSYCLKernelWithIL(Context, IL));
+  }
+
+  { // Compile a trivial CM kernel.
+    std::cout << "Test case3\n";
+    online_compiler<source_language::cm> Compiler;
+    try {
+      std::vector<byte> IL = Compiler.compile(std::string(CMSource));
+
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+  }
+
+  { // Compile a source with syntax errors.
+    std::cout << "Test case4\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    bool TestPassed = false;
+    try {
+      IL = Compiler.compile(std::string(CLSourceSyntaxError));
+    } catch (sycl::exception &e) {
+      std::string Msg = e.what();
+      if (Msg.find("syntax error here") != std::string::npos)
+        TestPassed = true;
+      else
+        std::cerr << "Unexpected exception: " << Msg << "\n";
+    }
+    assert(TestPassed && "Failed to throw an exception for syntax error");
+  }
+
+  { // Compile a good CL source using unrecognized compilation options.
+    std::cout << "Test case5\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    bool TestPassed = false;
+    try {
+      IL = Compiler.compile(
+          std::string(CLSource),
+          // Intentionally use incorrect option.
+          std::vector<std::string>{std::string("WRONG_OPTION")});
+    } catch (sycl::exception &e) {
+      std::string Msg = e.what();
+      if (Msg.find("WRONG_OPTION") != std::string::npos)
+        TestPassed = true;
+      else
+        std::cerr << "Unexpected exception: " << Msg << "\n";
+    }
+    assert(TestPassed &&
```

Comment:
same as above

Reply:
looks like I haven't pushed the 'add' button :) My concern was that `assert` is nop if the test is built w/o assertions enabled, in which case the check is not performed. I suggest to refactor not to rely on assert being not a no-op.

Reply:
Fixed in 392cc468cb3b1d41a23c32b9546339c0b3c19057